### ISSUE TITLE
CPU: Cycle accurate timings for divs/divu

### ISSUE THREAD

**Issue by ggnkua** [stefanberndtsson/ostis#101]
Hi,

If I read the cpu emulation code correctly, divs/divu cycles are calculated simply by adding the worst case cycle count. If that's the case then the following code Jorge Qwik (ijor) will give accurate cycles for all cases of division:

http://pasti.fxatari.com/68kdocs/div68kCycleAccurate.c

**Comment by stefanberndtsson** [comment]
Yes, I know about that one, and I had some thoughts of implementing it, but at the moment, I can't due to ostis being LGPL, and that code being GPL. I won't rule out that this can be solved, but I won't actually do anything about it until I've started rewriting the cpu to be clocked instead of instruction-stepped as it is now. I'll keep this open though.

**Comment by larsbrinkhoff** [comment]
It looks like it might be possible to implement the hardware algorithm, and thus compute the result in parallel with the cycle time.  It would be a good fit for our close-to-the-hardware policy.

**Comment by larsbrinkhoff** [comment]
The Yacht table also seems to have some algorithm for calculating the number of `DIVU` and `DIVS` cycles.
